### DISCUSSION TITLE
wikijs - saving a page - error - Cannot read property 'writeFile' of null

### DISCUSSION THREAD

**Discussion by pcm77** [requarks/wiki#3268] [Error / Bug Report]
I've built wikijs (version: 2.5.170) on a raspberry pi 4 (using the latest respberry pi OS: Release date: August 20th 2020
Kernel version: 5.4). I'm not using docker. The install is on the native OS.

I'm using node -v: 10.21.0
I'm using postgres psql (11.9 (Raspbian 11.9-0+deb10u1))

I built a standard version of wikijs and it looks fine. Then I restored my old wikijs postgres database which was previously hosted on ubuntu server 20.04. The restore appeared to work fine and all pages from my backup are available.

However, when a create a new page and then subsequently save, the following error message is displayed
"Cannot read property 'writeFile' of null"

However, the page saves but a conflict is reported and then the following error message is displayed:
"A more recent version of this page was saved by Administrator, a few seconds ago."

Does anyone have any idea why new pages are not saving correctly?

Thanks
Peter.

**Comment by ghost** [comment]
You haven't provided the required info about your host! (OS, Wiki.js version, Database engine)

**Comment by pcm77** [comment]
OS: Raspberry Pi OS Release date: August 20th 2020, Kernel version: 5.4)
Wiki.js version: 2.5.170
Database engine: Postgres (11.9 (Raspbian 11.9-0+deb10u1))

**Comment by pcm77** [comment]
Hello,
I have an update.
I've built a brand new Ubuntu server 20.4.01 which is exactly the same OS build as the original machine. I basically wanted to eliminate the raspberry pi OS/hardware from the troubleshooting.

From the original server, I backup the database using the following command:
pg_dump wiki -U wiki -F c > wikibackup.dump

I then SFTP to the target machine.

I then copy to the new directory ready for import:
mv /home/peter/wikibackup.dump /var/lib/postgresql/

I then do the restore:
sudo -i -u postgres
pg_restore -C -d wiki_db wikibackup.dump

As before, the wiki is restored. All pages from the original are accessible.
However, when I try to update a page I get the following errors in the log file:

2020-12-03T14:30:55.564Z [MASTER] info: (STORAGE/SFTP) Updating file 01_builds/rp/wiki-js...
2020-12-03T14:30:55.565Z [MASTER] warn: Cannot read property 'writeFile' of null

Any help would be appreciated. :)

thank you in advance

**Comment by PythonCoderAS** [comment]
I looked into the bug and it seems to occur with SFTP when you cannot ssh into the respective host. Try typing `ssh <target>` from the server running the wiki and see if it succeeds. In my case it was failing because I messed with DNS and my target was no longer resolving, causing the error to happen.
